src/renderer/views/wallet/CreateView/PhraseView.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react'
+
+import { useHistory } from 'react-router'
+
+import MnemonicConfirmScreen from '../../../components/wallet/NewMnemonicConfirm'
+import MnemonicGenerate from '../../../components/wallet/NewMnemonicGenerate'
+import { useWalletContext } from '../../../contexts/WalletContext'
+import * as walletRoutes from '../../../routes/wallet'
+
+export const PhraseView: React.FC = () => {
+  const [mnemonicInfo, setMnemonicInfo] = useState<{ phrase: string; password: string } | undefined>()
+  const history = useHistory()
+  const { keystoreService } = useWalletContext()
+
+  if (!mnemonicInfo) {
+    return <MnemonicGenerate onSubmit={setMnemonicInfo} />
+  }
+
+  return (
+    <MnemonicConfirmScreen
+      mnemonic={mnemonicInfo.phrase}
+      onConfirm={() => {
+        keystoreService.addKeystore(mnemonicInfo.phrase, mnemonicInfo.password).then(() => {
```

Comment:
Do we need `then` handler here? I guess `WalletView` would handle it by changing its routes after adding a keystore.

Reply:
@veado  unfortunately yes. when form is submitted it actually triggers `keystore$` which triggers `WalletView`. but... the history's route is still `/wallet/create` and it still renders `CreateView`. there is some logic at the `WalletView` based on 'is there a keystore?' and we can not just return any `Redirect` when `isLocked === false`. so here we have to redirect a user once to some base route. and also we have to do it exactly AFTER `keystore$` is updated  (at the then handler) and info about a keystore is available to render appropriate page

Reply:
In `ImportPhrase` we do same `keystoreService.addKeystore`, but handle possible errors there: https://github.com/thorchain/asgardex-electron/blob/81dfc0f751897d74d7a704f1e5d698a80ed5d234/src/renderer/components/wallet/ImportPhrase.tsx#L76-L80 

Could you try to handle errors and progress state in `MnemonicConfirmScreen` similar as we have it in `ImportPhrase`?

Reply:
@veado mice idea. added error handling but not sure here about loading state. we are not blocked by network requests here and adding a keystore here is just a writing ti the file on the local machine. as it really fast i believe we may ignore it and just handle ad error. dont you think so ?

Reply:
And we have a call of [`encryptToKeyStore` of `asgardex-crypto`](https://gitlab.com/thorchain/asgardex-common/asgardex-crypto/-/blob/master/src/crypto.ts#L70-108) in `addKeystore, which might take some time. Agree, if it's really that fast as expected, we don't need to give any "importing ..." feedback. Just let's check it out before.

Reply:
@veado yep. it so fast as expected even in dev mode. it switches to wallet view like Flash ⚡

Reply:
Cool, then let skip a `progress` message as you suggested.